addons/html_builder/static/src/builder/snippets_menu.js

```diff
@@ -210,9 +212,28 @@ export class SnippetsMenu extends Component {
 
     onBeforeUnload(event) {
         if (!this.isSaving && this.editor.shared.dirty.isEditableDirty()) {
+            event.preventDefault();
             event.returnValue = "Unsaved changes";
         }
     }
+
+    async onBeforeLeave() {
```

Comment:
Note: I did not find a better way to do but when clicking on the "back" arrow to trigger `beforeLeave`, it seems that the browser history is already updated even before clicking on "confirm" or "cancelled" in the dialog. This problem also appears in the settings page (see `settings_form_controller.js`).